webapp/src/components/CreateMeetingNoteForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { Alert, Card, CardContent, Snackbar, TextField } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import React, { FormEventHandler, useState, useContext } from 'react';
+
+import { EntityId } from '../types/api';
+import SessionContext from './SessionContext';
+
+interface CreateMeetingNoteFormProps {
+  meetingId: EntityId;
+  onMeetingNoteChanged?: (id: EntityId) => void;
+  greatestSortOrderOfMeetingNotesList: number;
+}
+
+const CreateMeetingNoteForm = ({
+  meetingId,
+  onMeetingNoteChanged,
+  greatestSortOrderOfMeetingNotesList,
+}: CreateMeetingNoteFormProps) => {
+  const { principalId } = useContext(SessionContext);
+
+  const [isSavingMeetingNote, setIsSavingMeetingNote] = useState(false);
+  const [saveMeetingNoteError, setSaveMeetingNoteError] = useState(null);
+  const [isSuccessMessageVisible, setIsSuccessMessageVisible] = useState(false);
+  const [meetingNoteText, setMeetingNoteText] = useState('');
+  const onSubmit: FormEventHandler = event => {
+    event.preventDefault();
+    setIsSavingMeetingNote(true);
+
+    fetch(`${process.env.REACT_APP_API_ORIGIN}/meetings/${meetingId}/notes`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        agenda_owning_participant_id: principalId,
+        note_text: meetingNoteText,
+        sort_order: greatestSortOrderOfMeetingNotesList + 1,
```

Comment:
Why not compute this before passing it in? Then the prop can just be called `sortOrder`.

Reply:
Thanks.  The new changes pass down the correct sort order to the form as opposed to computing the value in the form.